src/anemoi/inference/runner.py

```diff
@@ -376,6 +379,32 @@ def run(
                     )
                     raise ValueError(f"Field '{name}' has NaNs and is not marked as imputable")
 
+        global_rank = int(
+            os.getenv("SLURM_PROCID", 0)
+        )  # Get rank of the current process, equivalent to dist.get_rank()
+        world_size = int(os.getenv("SLURM_NTASKS", 1))  # Total number of processes
+        local_rank = int(os.getenv("SLURM_LOCALID", 0))  # Get GPU num of current process
+        if global_rank == 0:
+            LOGGER.info("World size: %d", world_size)
+        addr = os.getenv("MASTER_ADDR", "localhost")  # localhost should be sufficient to run on a single node
+        port = os.getenv("MASTER_PORT", 10000 + random.randint(0, 10000))  # random port between 10,000 and 20,000
```

Comment:
Does the fallback to `10000 + random.randint(0, 10000)` work without consistent seeding across ranks? Maybe we should use `SLURM_JOBID` instead?